### ISSUE TITLE
Semi-supervised adversarial autoencoder network

### ISSUE THREAD

**Issue by SASVDERDBGTYS** [SASVDERDBGTYS/BCGNET_CNN#3]
Create a working adversarial autoencoder network consisting of a autoencoder network followed by a discriminator network. The input to the network should be corrupted EEG data from all 63 channels and the goal would be to autoencoder would be to output BCG contained in the 63 channels. Then a subtraction is performed between the corrupted EEG and the predicted BCG to obtain the predicted cleaned EEG. The predicted cleaned EEG is then fed into the discriminator network together with the ground truth clean EEG signal and a loss value is subsequently computed and backpropagation is performed. 

Note: The current optimal networks cnn_arch_015 and cnn_arch_020 already has a shape that's very similar to an autoencoder. 

Reference:
https://arxiv.org/abs/1511.05644

(Some errors?) (Should be two discriminator functions rather than just 1)
https://github.com/alimirzaei/adverserial-autoencoder-keras/blob/master/SSAAE.py

(Note that here the discriminator is set to untrainable)
https://github.com/eriklindernoren/Keras-GAN/blob/636ec0533df1d1ba2bfe4ec9ae7aa66bd7ee2177/aae/aae.py#L30

https://towardsdatascience.com/a-wizards-guide-to-adversarial-autoencoders-part-4-classify-mnist-using-1000-labels-2ca08071f95

**Comment by SASVDERDBGTYS** [comment]
Before trying out the adversarial part. Firstly, try to create an variational autoencoder with a structure that's similar to the current optimal networks cnn_arch_015 and cnn_arch_020 and subsequently experiment with the dimension of the latent space and see what happens when that dimension is reduced (will the output be BCG since that's the largest source of variance or something else).

**Comment by SASVDERDBGTYS** [comment]
Just realized but for the previous commit there could be a branching off from the middle dense layer and one path would be reduced into the latent space and one path would lead to BCG prediction following the current architecture since functional API supports branching off.

**Comment by dy2406** [comment]
Yes, I did do this branching off to have the latent space and BCG prediction in my other (more updated) file. I thought this would just look cleaner for only autoencoder training, since the branching would only be needed for the discriminators.

**Comment by SASVDERDBGTYS** [comment]
Migrating to the other github repo, closing the issue
